### PR TITLE
Feature/get events

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What
- Add endpoint allowing events to be retrieved for a collection
- Check that collection exists before getting events
- Add component test scenario to GET collections when none exist

### How to review
Review changes / tests

### Who can review
Anyone

### CODE REVIEW

File: features/get_events.feature [hunk 1]
```diff
@@ -0,0 +1,60 @@
+Feature: Get Collection events
```
Comment (nshumoogum) [inline features/get_events.feature:1]: Do we not write feature test for internal server error - failure to connect to database?

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: features/get_events.feature (hunk 1 above)
Comment (CarlHembrough) [reply features/get_events.feature:1]: Good question - I had leaned towards more "happy path" scenarios for the component tests, but I guess it could be argued that more error scenarios could be added. It's also a tradeoff with what we already cover in the unit tests. It's also made a bit more tricky by having the in memory MongoDB, as the error test would have to be configured differently.

Comment (nshumoogum) [reply features/get_events.feature:1]: I think this is something to come back to. The duplication of test coverage between the component tests and the unit tests should be removed once we have a better understanding of the overlap and the coverage from both approaches - the theory would be the component tests should cover the handler tests where we are dealing with the request endpoints but any business logic like validation and pagination should be covered by unit tests

**Review by nshumoogum — APPROVED** [review]
